ecommerce/frontend/src/components/ShoppingCart/ShoppingCartPage.js

```diff
@@ -57,8 +57,8 @@ export const ShoppingCartPage = ({ currency = 'TL' }) => {
                     </Row>
                 </div>
                 <div className="shopping-proceed-button">
-                    <Button type="primary" onClick={onCheckout} block>
-                        Proceed to payment
+                    <Button type="primary" onClick={onCheckout} block disabled={shoppingCart?.length === 0}>
+                        <Link to="/checkout">Proceed to payment</Link>
                     </Button>
```

Comment:
not sure if it makes sense to have both an onClick and a link? if you have complex logic then it is enough to have onClick which checks for some things and if okay uses history.push(url) to go to /checkout